mlir/lib/Dialect/Math/Transforms/PolynomialApproximation.cpp

```diff
@@ -1316,6 +1358,106 @@ LogicalResult SinAndCosApproximation<isSine, OpTy>::matchAndRewrite(
   return success();
 }
 
+//----------------------------------------------------------------------------//
+// SinhOp and CoshOp approximation.
+//----------------------------------------------------------------------------//
+
+namespace {
+
+template <bool isSine, typename OpTy>
```

Comment:
There is no need to pass the template parameter `isSine`. It can be deduced later.